client/views/admin/settings/groups/voip/RemoveAgentButton.tsx

```diff
@@ -1,28 +1,42 @@
 import { Table, Icon, Button } from '@rocket.chat/fuselage';
 import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
-import React, { FC } from 'react';
+import React, { FC, useEffect } from 'react';
 
 import GenericModal from '../../../../../components/GenericModal';
 import { useSetModal } from '../../../../../contexts/ModalContext';
-import { useEndpoint } from '../../../../../contexts/ServerContext';
+import { useEndpoint, useStream } from '../../../../../contexts/ServerContext';
 import { useToastMessageDispatch } from '../../../../../contexts/ToastMessagesContext';
 import { useTranslation } from '../../../../../contexts/TranslationContext';
+import { useUserId } from '../../../../../contexts/UserContext';
 
-const RemoveAgentButton: FC<{ username: string; reload: () => void }> = ({ username, reload }) => {
+const RemoveAgentButton: FC<{ username: string; reload: () => void; extension: string }> = ({ username, reload, extension }) => {
 	const removeAgent = useEndpoint('DELETE', 'omnichannel/agent/extension');
 	const setModal = useSetModal();
 	const dispatchToastMessage = useToastMessageDispatch();
 	const t = useTranslation();
+	const subscribeToNotifyUser = useStream('notify-user');
+	const userId = useUserId();
 
 	const handleRemoveClick = useMutableCallback(async () => {
 		try {
 			await removeAgent({ username });
 		} catch (error: any) {
 			dispatchToastMessage({ type: 'error', message: error });
 		}
-		reload();
 	});
 
+	useEffect(() => {
```

Comment:
Instead  of a useEffect here, maybe you can add this action to to the `handleRemoveClick` or create a mutableCallback

Reply:
Since the extension state change happens from an action done by the user, It doesn't make much sense to listen to changes and then to the action, is it better to act when the event happens